src/dotnet/Fable.Compiler/Transforms/FSharp2Fable.fs

```diff
@@ -227,6 +227,51 @@ let private transformExpr (com: IFableCompiler) (ctx: Context) fsExpr =
+                                                    (BasicPatterns.TypeTest(tt, BasicPatterns.Value vv), _, _)), _))
+            when var.FullName = "matchValue" && memb.FullName = "Microsoft.FSharp.Core.Operators.box"
+                && vv.FullName = "matchValue" && (getFsTypeFullName tt) = "System.IFormattable" ->
+        let r, typ = makeRangeFrom fsExpr, makeType com ctx.GenericArgs fsExpr.Type
+        let argTypes = membArgTypes |> List.map (makeType com ctx.GenericArgs)
+        let! args = transformExprList com ctx membArgs
+        let entity = memb.DeclaringEntity.Value
+        let opName = "ToString"
+        let membOpt = tryFindMember com entity ctx.GenericArgs opName false argTypes
+        return (match membOpt with
+                | Some memb -> makeCallFrom com ctx r typ false argTypes None args memb
+                | None -> failwithf "Cannot find member %A.%A" (entity.FullName) opName)
+
+    // work-around for optimized hash operator (Operators.hash)
+    | BasicPatterns.Call(Some expr, memb, _, [], [BasicPatterns.Call(None, comp, [], [], [])])
+            when memb.FullName.EndsWith(".GetHashCode") &&
+                 comp.FullName = "Microsoft.FSharp.Core.LanguagePrimitives.GenericEqualityERComparer" ->
+        let r, typ = makeRangeFrom fsExpr, makeType com ctx.GenericArgs fsExpr.Type
+        let argTypes = [expr.Type] |> List.map (makeType com ctx.GenericArgs)
+        let! args = transformExprList com ctx [expr]
+        let entity = comp.DeclaringEntity.Value
+        let opName = "GenericHash"
+        let membOpt = tryFindMember com entity ctx.GenericArgs opName false argTypes
+        return (match membOpt with
+                | Some memb -> makeCallFrom com ctx r typ false argTypes None args memb
+                | None -> failwithf "Cannot find member %A.%A" (entity.FullName) opName)
+
+    // work-around for optimized equality operator (Operators.(=))
+    | BasicPatterns.Call(Some e1, memb, _, [], [BasicPatterns.Coerce (t2, e2); BasicPatterns.Call(None, comp, [], [], [])])
+            when memb.FullName.EndsWith(".Equals") && t2.HasTypeDefinition && t2.TypeDefinition.CompiledName = "obj" &&
+                 comp.FullName = "Microsoft.FSharp.Core.LanguagePrimitives.GenericEqualityComparer" ->
+        let r, typ = makeRangeFrom fsExpr, makeType com ctx.GenericArgs fsExpr.Type
+        let argTypes = [e1.Type; e2.Type] |> List.map (makeType com ctx.GenericArgs)
+        let! args = transformExprList com ctx [e1; e2]
+        let entity = comp.DeclaringEntity.Value
+        let opName = "GenericEquality"
+        let membOpt = tryFindMember com entity ctx.GenericArgs opName false argTypes
+        return (match membOpt with
+                | Some memb -> makeCallFrom com ctx r typ false argTypes None args memb
+                | None -> failwithf "Cannot find member %A.%A" (entity.FullName) opName)
```

Comment:
It'd be great if we could move these workarounds some where else. For Fable 2 I was trying to avoid dealing with optimizations in this pattern match. I'll see what can be done later :)